### PR TITLE
pymol: Update to 2.5.0 and remove X11 and python27 variants

### PR DESCRIPTION
**Pull request by jwhowarth**
pymol: Update to the 2.5.0 release as well as remove the deprecated X11 and python27 variants

###### Type(s)

- [ ] bugfix
- [ x] enhancement
- [ ] security fix

###### Tested on

macOS 11.4
Xcode 12.5

###### Verification 
Have you

- [ x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ x] checked your Portfile with `port lint`?
- [ x] tried existing tests with `sudo port test`?
- [x ] tried a full install with `sudo port -vst install`?
- [ x] tested basic functionality of all binary files?

### CODE REVIEW

File: science/pymol/Portfile [hunk 1]
```diff
@@ -5,10 +5,9 @@ PortGroup           python 1.0
 PortGroup           active_variants 1.1
```
Comment (reneeotten) [inline science/pymol/Portfile:5]: please remove this PG as it is no longer used

File: science/pymol/Portfile [hunk 2]
```diff
@@ -23,46 +22,34 @@ long_description    PyMOL is a molecular graphics system with an embedded Python
...
+    checksums       rmd160  6432d47616aee1e5d9d40cc2a74425b33159d7a0 \
+                    sha256  67c5aa944ce9ac2e3babd42e7e2b78ed11719283a2c0a063e100cfe24665c0d3 \
+                    size    10471287
 }
 
-variant python27 conflicts python36 python37 python38 python39 description {Use Python 2.7} {}
 variant python36 conflicts python27 python37 python38 python39 description {Use Python 3.6} {}
 variant python37 conflicts python27 python36 python38 python39 description {Use Python 3.7} {}
 variant python38 conflicts python27 python36 python37 python39 description {Use Python 3.8} {}
 variant python39 conflicts python27 python36 python37 python38 description {Use Python 3.9} {}
```
Comment (reneeotten) [inline science/pymol/Portfile:43]: the conflict with `python27` should be removed here as this variant doesn't exist anymore

File: science/pymol/Portfile [hunk 3]
```diff
@@ -23,46 +22,34 @@ long_description    PyMOL is a molecular graphics system with an embedded Python
...
     version         20210428-[string range ${github.version} 0 7]
     conflicts       pymol
     revision        0
     maintainers     {reneeotten @reneeotten} openmaintainer
-    checksums       rmd160  b8f596935a043a3ecde7b8f0e1c914b74277d1d0 \
-                    sha256  92f39021487fa1c3da1840721c7ad4c3992dfcb54d73fef27fb30842828425a8 \
-                    size    10469954
+    checksums       rmd160  6432d47616aee1e5d9d40cc2a74425b33159d7a0 \
+                    sha256  67c5aa944ce9ac2e3babd42e7e2b78ed11719283a2c0a063e100cfe24665c0d3 \
+                    size    10471287
```
Comment (reneeotten) [inline science/pymol/Portfile:37]: you should only update the `checksums` for the "main" `pymol` port; this is still using the same commit so the `checksums` haven't changed

File: science/pymol/Portfile [hunk 4]
```diff
@@ -80,44 +67,17 @@ depends_lib-append  port:freetype \
...
 
 if {${subport} eq "pymol-devel"} {
-    patchfiles-delete \
-                    apbs-psize.patch  \
-                    python_string_split.patch \
-                    pdb2pqr.patch \
     patchfiles-replace \
                     setup.py.diff \
                     patch-devel-setup.py.diff
 }
```
Comment (reneeotten) [inline science/pymol/Portfile:77]: the patchfiles should be the same now for both `pymol` and `pymol-devel` so you can remove the special case for the `pymol-devel` port. 

However, the `setup.py.diff` doesn't apply anymore (how did you build this locally?) - you should probably just use rename the `patch-devel-setup.py.diff` to `patch-setup.py.diff` and use that one instead.

File: science/pymol/Portfile [hunk 5]
```diff
@@ -80,44 +67,17 @@ depends_lib-append  port:freetype \
...
-                    apbs-psize.patch  \
-                    python_string_split.patch \
-                    pdb2pqr.patch \
                     setup.py.diff
 
 if {${subport} eq "pymol-devel"} {
-    patchfiles-delete \
-                    apbs-psize.patch  \
-                    python_string_split.patch \
-                    pdb2pqr.patch \
```
Comment (reneeotten) [inline science/pymol/Portfile:110]: please also remove the `pdb2pqr.patch` from the `files` directory

File: science/pymol/Portfile [hunk 6]
```diff
@@ -80,44 +67,17 @@ depends_lib-append  port:freetype \
...
-    require_active_variants tk "" quartz
-
-    build.args-append   --glut --no-osx-frameworks
-}
-
 patchfiles          pymol_shell.diff \
-                    pmg_tk_platform.patch \
-                    apbs-psize.patch  \
-                    python_string_split.patch \
-                    pdb2pqr.patch \
```
Comment (reneeotten) [inline science/pymol/Portfile:103]: please also remove the `pdb2pqr.patch` from the `files` directory

Comment (reneeotten) [inline science/pymol/Portfile:103]: please also remove the `pdb2pqr.patch` from the `files` directory

File: science/pymol/Portfile [hunk 7]
```diff
@@ -23,46 +22,34 @@ long_description    PyMOL is a molecular graphics system with an embedded Python
...
-    python.default_version 27
-} elseif {[variant_isset python36]} {
+if {[variant_isset python36]} {
     python.default_version 36
 } elseif {[variant_isset python37]} {
     python.default_version 37
 } elseif {[variant_isset python38]} {
     python.default_version 38
 } elseif {[variant_isset python39]} {
     python.default_version 39
```
Comment (reneeotten) [inline science/pymol/Portfile:52]: right now there is no default variant set anymore. The last two lines should be changed to:
```
} else {
    default_variants +python39
     python.default_version 39
```

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks for the update @jwhowarth! There are a few issue with it though and I've left some review comments below.

Also, the test don't run right now (it looks like the are not part of the download anymore) so that should either be fixed or the test phase should be removed.

**Comment by jwhowarth** [comment]
Replaced with pull request #11158

**Pull request closed without merging**